Testscripts/Linux/ETHTOOL-OFFLOADING-SETTING.sh

```diff
@@ -0,0 +1,268 @@
+
+declare -A offload_features
+offload_features=([tcp-segmentation-offload]=tso [udp-fragmentation-offload]=ufo [generic-segmentation-offload]=gso [generic-receive-offload]=gro [large-receive-offload]=lro [rx-vlan-offload]=rxvlan [tx-vlan-offload]=txvlan [rx-checksumming]=rx [tx-checksumming]=tx)
+
+__iterator=1
+__ip_iterator_1=1
+__ip_iterator_2=2
+
+# Check all the interfaces with VF
+while [ $__iterator -le "$vf_count" ]; do
+    # Extract VF_IP values
+    ip_variable_name="VF_IP$__ip_iterator_1"
+    static_IP_1="${!ip_variable_name}"
+    ip_variable_name="VF_IP$__ip_iterator_2"
+    static_IP_2="${!ip_variable_name}"
+
+    synthetic_interface_vm_1=$(ip addr | grep $static_IP_1 | awk '{print $NF}')
+    LogMsg  "Synthetic interface found: $synthetic_interface_vm_1"
+    if [[ $DISTRO_VERSION =~ ^6\. ]]; then
+        synthetic_MAC=$(ip link show ${synthetic_interface_vm_1} | grep ether | awk '{print $2}')
+        vf_interface_vm_1=$(grep -il ${synthetic_MAC} /sys/class/net/*/address | grep -v $synthetic_interface_vm_1 | sed 's/\// /g' | awk '{print $4}')
+    else
+        vf_interface_vm_1=$(find /sys/devices/* -name "*${synthetic_interface_vm_1}" | grep "pci" | sed 's/\// /g' | awk '{print $12}')
+    fi
+    LogMsg "Virtual function found: $vf_interface_vm_1"
+
+    # Ping the remote host
+    i=0
+    ping_exit_code=1
+    while [ $i -lt 4 ]
+    do
+        i=$(($i+1))
+        sleep 10
+        ping -c 11 "$static_IP_2" > "$static_IP_2.log" 2>&1
+        ping_exit_code=$?
+        LogMsg "Attempt #$i to ping $VF_IP2 through $synthetic_interface_vm_1"
+        if [ "$ping_exit_code" == 0 ]; then
+            LogMsg "Successfully pinged $VF_IP2 through $synthetic_interface_vm_1"
+            break
+        fi
+    done
+
+    if [ "$ping_exit_code" == 1 ]; then
+        LogErr "Unable to ping $VF_IP2 through $synthetic_interface_vm_1"
+        SetTestStateFailed
+        exit 0
+    fi
+
+    # Take a note on the offloading state and set it on
+    for key in ${!offload_features[*]}
+    do
+        on_or_off=$(ethtool -k $synthetic_interface_vm_1 2>&1 | grep $key | awk {'print $2'})
+        fixed_or_not=$(ethtool -k $synthetic_interface_vm_1 2>&1 | grep $key | awk {'print $3'})
+        LogMsg "The initial status of $key for $synthetic_interface_vm_1: $on_or_off $fixed_or_not"
+        # Set the offload on if its status is off and not fixed
+        if [[ "$on_or_off" == "off" && x"$fixed_or_not" != x"[fixed]" ]];then
+            LogMsg "Set ${offload_features[$key]} on"
+            result=$(ethtool -K $synthetic_interface_vm_1 ${offload_features[$key]} on 2>&1)
+            if [[ "$result" = *"Could not change any device features"* ]]; then
+                LogErr "$result"
+                LogErr "The kernel doesn't support this feature"
+            fi
+        fi
+    done
+
+    tx_value=$(cat /sys/class/net/"${vf_interface_vm_1}"/statistics/tx_packets)
```

Comment:
In case, this path may not be available or accessible. Please filter or check 'tx_value'

Reply:
Fixed it.